### PR TITLE
Iox #743 fix windows c waitset example

### PR DESCRIPTION
**Pull request by elfenpiff**
## Pre-Review Checklist for the PR Author

1. [x] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [x] Tests follow the [best practice for testing][testing]
1. [x] Changelog updated [in the unreleased section][changelog] including API breaking changes
1. [x] Branch follows the naming format (`iox-#123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit messages are signed (`git commit -s`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/advanced/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/release-notes/iceoryx-unreleased.md

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
    - [x] Each unit test case has a unique UUID
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

- #743

### CODE REVIEW

File: iceoryx_examples/waitset_in_c/ice_c_waitset_timer_driven_execution.c [hunk 1]
```diff
@@ -72,7 +73,8 @@ void* cyclicTriggerCallback(void* dontCare)
 bool createThread(pthread_t* threadHandle, void* (*callback)(void*))
```
Comment (elBoberido) [inline iceoryx_examples/waitset_in_c/ice_c_waitset_timer_driven_execution.c:73]: Oh, I guess I forgot to change this to `int` after noticing that pthread_create also returns an int. Could you please change the return type?

**Comment by elBoberido** [comment]
Please change the target branch to master

**Review by elBoberido — DISMISSED** [review]

File: iceoryx_examples/waitset_in_c/ice_c_waitset_timer_driven_execution.c [hunk 2]
```diff
@@ -120,7 +117,7 @@ int main()
     // start a thread which triggers cyclicTrigger every second
     //! [cyclic trigger thread]
     pthread_t cyclicTriggerThread;
-    if (createThread(&cyclicTriggerThread, cyclicTriggerCallback))
+    if (!createThread(&cyclicTriggerThread, cyclicTriggerCallback))
```
Comment (elBoberido) [inline iceoryx_examples/waitset_in_c/ice_c_waitset_timer_driven_execution.c:120]: Can you please update the documentation and remove the note which tells that this does not run on Windows?

File: iceoryx_examples/waitset_in_c/ice_c_waitset_timer_driven_execution.c [hunk 3]
```diff
@@ -24,7 +24,8 @@
 #include "topic_data.h"
 
 #if defined(_WIN32)
-typedef long unsigned int pthread_t;
+#include <windows.h>
```
Comment (FerdinandSpitzschnueffler) [inline iceoryx_examples/waitset_in_c/ice_c_waitset_timer_driven_execution.c:27]: Please update the copyright year.

**Review by FerdinandSpitzschnueffler — APPROVED** [review]

**Pull request merged**